### DISCUSSION TITLE
Error in LIS Ensemble Smoothing Run for Data Assimilation of GRACE TWS into NoahMP 3.6 LSM Model: [ERR] ESMF_StateGet failed for sm1 in noahmp36_gettws Stopping.

### DISCUSSION THREAD

**Discussion by ThangjamSatrajitSingh** [NASA-LIS/LISF#1625] [Q&A] [answered]
Hi, I am currently working on data assimilation of GRACE TWS into the NoahMP 3.6 LSM model over India. For the LDT and LIS executables, I cloned and compiled the master version from GitHub. So far, the preliminary stages of the workflow have been successful. The following processes have all completed without any errors:

1. LDT Parameter Generation Run
2. LIS Open Loop (OL) Run
3. LDT Anomaly Correction Run

However, I am encountering an issue in the LIS Data Assimilation (DA) Ensemble Smoothing Run. The error message returned is as follows:
**[ERR] ESMF_StateGet failed for sm1 in noahmp36_gettws Stopping.**

I’ve attached the relevant configuration and log files in the link below:
[GRACE_DA_FILES_SHARE](https://csciitd-my.sharepoint.com/:f:/g/personal/satrajit_cstaff_iitd_ac_in/ElMvq92UCWJNu4IeCIcMp5EB0pbSoERjiIBWUCkckLcoqg?e=WxZ5zR)

I would greatly appreciate any guidance or insights you may have on resolving this issue. Thank you very much for your time and support in advance.

Best regards,
Thangjam Satrajit Singh

**Comment by dmocko** [comment]
Hi @ThangjamSatrajitSingh 

Can you please also share the lis.config file for the GRACE_DA run, as well as these files?
```
 ./DA_INPUT/clsm_tws_attribs.txt
 ./DA_INPUT/clsm_tws_pertattribs.txt
```

**Comment by dmocko** [comment]
@ThangjamSatrajitSingh 

Please also share these files that you have listed in your lis.config file as well:

```
Forcing variables list file:

Forcing attributes file:
Forcing perturbation attributes file:
State attributes file:
State perturbation attributes file:
Observation attributes file:
Observation perturbation attributes file:

Model output attributes file:
```

**Comment by dmocko** [comment]
I do see your lis.config file, but I will also need to see all of the files that I just listed.

**Reply by ThangjamSatrajitSingh** [reply]
Hi @dmocko,

I’ve attached the files here:  
[DA_SHARE_NEW.zip](https://github.com/user-attachments/files/17526688/DA_SHARE_NEW.zip)

The file names are as follows:

- **Forcing Variables List**: `forcing_variables.txt`
- **Forcing Attributes**: `forcing_attribs.txt`
- **Forcing Perturbation Attributes**: `forcing_pert_attribs.txt`
- **State Attributes**: `clsm_tws_attribs.txt`
- **State Perturbation Attributes**: `clsm_tws_pertattribs.txt`
- **Observation Attributes**: `GRACEtws_attribs.txt`
- **Observation Perturbation Attributes**: `GRACEtws_pertattribs.txt`
- **Model Output Attributes**: `MODEL_OUTPUT_LIST.TBL`

Thank you very much for your time and help with this!

Best regards,  
Thangjam Satrajit Singh

**Comment by ThangjamSatrajitSingh** [comment]
Hello @dmocko ,

I hope you’re doing well! I wanted to kindly follow up on this issue as I’ve tried running it with both NoahMP 3.6 and NoahMP 4.0.1, but I’m encountering the same error in both cases as follows:

- In noahmp.3.6   : `ESMF_StateGet failed for sm1 in noahmp36_gettws Stopping.   `
- In noahmp.4.0.1 : `ESMF_StateGet failed for sm1 in noahmp401_gettws Stopping.`

I have also attached the relevant files and logfiles for the noahmp.4.0.1 run here: 
[GRACE_DA_noahmp401.zip](https://github.com/user-attachments/files/17619573/GRACE_DA_noahmp401.zip)

If you have any suggestions or insights, I’d really appreciate it. Please let me know if there’s any other information I can provide.

Thank you for your time and assistance!

Best regards,
Thangjam Satrajit Singh

**Reply by dmocko** [reply, marked as answer]
Hi @ThangjamSatrajitSingh 

Thanks for your patience.  We are very busy with our own model development and delivery, and along with meetings,
we haven't had time to follow up on this.

Thanks for sending all the relevant files.  I think I have a solution for you.

First, I did see in your lis.config that you are wondering if setting `LAI data source:` to "none" was part of the problem.  It is not.
You are choosing the vegetation physics option "4", which will use the annual maximum value at each grid point of the monthly climatology of greenness from the lis_input file produced by LDT, along with the LAI data in the MPTABLE file.  So, it does not use/need LAI data from the LDT file.

The issue is that you are using state perturbation attributes and perturbation files specific to the Catchment LSM, and not to Noah-MP.  Please use the files that I have attached, and let us know if it works.

[noahmp_tws_attribs.txt](https://github.com/user-attachments/files/17639226/noahmp_tws_attribs.txt)
[noahmp_tws_pertattribs.txt](https://github.com/user-attachments/files/17639227/noahmp_tws_pertattribs.txt)

The error message that you are seeing is the code letting you know that the code can't find "Soil Moisture Layer 1", which is in the noahmp_tws_attribs.txt file.

**Reply by msaharia** [reply]
Thank you @dmocko! We appreciate it. 

~ Manabendra

**Comment by ThangjamSatrajitSingh** [comment]
Hi @dmocko,
  
  Thank you so much for getting back to me despite your busy schedule—I really appreciate it!
  Also thanks for sharing the `correct noahmp_tws_attribs.txt` and `noahmp_tws_pertattribs.txt` files. I’ll implement these updates and give it another try. I'll let you know if it resolves the issue or if any further problems arise.
  
  Thanks again for your time and support!
  
  Best regards,
  Thangjam Satrajit Singh

**Comment by msaharia** [comment]
Thank you, @dmocko - we have been able to successfully run it.

![CleanShot 2024-11-07 at 2  51 42@2x](https://github.com/user-attachments/assets/f5c6c6e4-aa4d-4828-a168-559ba9bc819e)

**Reply by ThangjamSatrajitSingh** [reply]
@dmocko - Thanks for your help on this!
